### PR TITLE
Redesign admin form

### PR DESCRIPTION
**Pull request by kiranparajuli589**
### Description
With this PR, the admin form is refactored with three different forms:
- OpenProject server host
- OpenProject OAuth Client Values
- Nextcloud OAuth Client Values

These forms render with three different modes:
- Disabled
- View
- Edit

according to the state of the integration system

Fixes: https://community.openproject.org/projects/nextcloud-integration/work_packages/42730
Fixes: https://community.openproject.org/projects/nextcloud-integration/work_packages/42615
Part of: https://community.openproject.org/projects/nextcloud-integration/work_packages/42862

Signed-off-by: Kiran Parajuli <kiranparajuli589@gmail.com>

### CODE REVIEW

**Comment by kiranparajuli589** [comment]
Todo after this PR:
- use separate variables to remember the admin settings so that if we cancel the edit form, we can set them back to as it was before edit.

File: lib/Controller/ConfigController.php [hunk 1]
```diff
@@ -144,14 +144,14 @@ public function setAdminConfig(array $values): DataResponse {
...
-			$this->config->deleteAppValue(Application::APP_ID, 'nc_oauth_client_id');
-		}
+//		if (isset($values['oauth_instance_url'])) {
+//			$oauthClientInternalId = $this->config->getAppValue(Application::APP_ID, 'nc_oauth_client_id', '');
+//			if ($oauthClientInternalId !== '') {
+//				$id = (int) $oauthClientInternalId;
+//				$this->oauthService->deleteClient($id);
+//			}
+//			$this->config->deleteAppValue(Application::APP_ID, 'nc_oauth_client_id');
+//		}
```
[the remark this answers is not in the document]
Comment (individual-it) [reply lib/Controller/ConfigController.php:154]: Leave the current behavior for this PR, we can implement the desired behavior in a separate PR

[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply lib/Controller/ConfigController.php]: reverted back to as it was. but still needs the readjustments.

File: src/components/admin/TextInput.vue [hunk 2]
```diff
@@ -39,6 +39,7 @@ import { translate as t } from '@nextcloud/l10n'
 export default {
 	name: 'TextInput',
 	props: {
+		// eslint-disable-next-line vue/require-prop-types
 		value: {
 			required: true,
 		},
```
Comment (individual-it) [inline src/components/admin/TextInput.vue:45]: why not set a default value here?

**Comment by kiranparajuli589** [comment]
A new work package is created [here](https://community.openproject.org/projects/nextcloud-integration/work_packages/42862) for these remaining todos. @individual-it @SwikritiT we can merge this one to move forward.

**Review by SwikritiT — COMMENTED** [review]
LGTM

File: src/components/AdminSettings.vue [hunk 3]
```diff
@@ -1,203 +1,418 @@
...
+			return this.formMode.ncOauth === F_MODES.EDIT
+		},
+		isOPOAuthFormInDisableMode() {
+			return this.formMode.opOauth === F_MODES.DISABLE
+		},
+		isNcOAuthFormInDisableMode() {
+			return this.formMode.ncOauth === F_MODES.DISABLE
+		},
+		adminFileStorageHref() {
+			return this.state.oauth_instance_url + '/admin/settings/storages'
```
Comment (individual-it) [inline src/components/AdminSettings.vue:223]: if the main OpenProject host has a trailing `/` this link will have two `//` that might give sometimes problems

**Review by individual-it — CHANGES_REQUESTED** [review]
great work!
just some minor comments


the second step does not need the `Copy Values` buttons
image

File: src/components/AdminSettings.vue [hunk 4]
```diff
@@ -1,203 +1,418 @@
...
-		</p>
-		<div class="grid-form">
-			<label for="openproject-oauth-instance">
-				<a class="icon icon-link" />
-				{{ t('integration_openproject', 'OpenProject instance address') }}
-			</label>
-			<input id="openproject-oauth-instance"
+		<div class="openproject-server-host">
+			<FormHeading index="1"
+				title="OpenProject Server"
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/AdminSettings.vue:6]: they are translated inside the component `FormHeading`

File: src/components/AdminSettings.vue [hunk 5]
```diff
@@ -1,203 +1,418 @@
...
+		adminFileStorageHref() {
+			return this.state.oauth_instance_url + '/admin/settings/storages'
+		},
+		openProjectClientHint() {
+			return this.translate('Go to your OpenProject')
+				+ ` <a class="link" href="${this.adminFileStorageHref}" target="_blank"`
+				+ 'title="Administration > File storages">'
+				+ this.translate('Administration > File storages')
+				+ '</a> '
+				+ this.translate('as an Administrator and start the setup and copy the values here.')
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply src/components/AdminSettings.vue]: used single translate

File: src/components/AdminSettings.vue (hunk 3 above)
Comment (kiranparajuli589) [reply src/components/AdminSettings.vue]: fixed

File: src/components/AdminSettings.vue [hunk 6]
```diff
@@ -120,7 +118,7 @@
 					:value="ncClientSecret" />
 				<Button v-if="isNcOAuthFormInEdit"
 					class="submit-btn"
-					text="Done"
+					text="Yes, I have copied these values"
```
Comment (individual-it) [inline src/components/AdminSettings.vue:121]: need translation

**Review by individual-it — APPROVED** [review]
I think some snapshots in the tests need to be recreated, rest looks good to me

**Pull request merged**